update/update.go

```diff
@@ -39,6 +51,33 @@ func New(config Config) (*Update, error) {
 func (u *Update) Test(ctx context.Context) error {
 	var err error
 
+	{
+		u.logger.LogCtx(ctx, "level", "debug", "message", "waiting for the guest cluster to be created")
+
+		o := func() error {
+			s, err := u.provider.CurrentStatus()
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			if s.HasCreatedCondition() {
+				return backoff.Permanent(microerror.Mask(hasDesiredStatusError))
```

Comment:
Why not just returning `nil` here?

Reply:
Oh wow. Captain obvious. I will provide a PR. Thanks.

Reply:
See https://github.com/giantswarm/e2etests/pull/9.